sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -0,0 +1,309 @@
+should not cache the data at level `L` in the memory hierarchy.
+a|
+[source]
+----
+cache_control_read<cache_control_read_type::streaming, L>
+----
+|
+This property requests that loads from memory through the `annotated_ptr`
+should cache the data at cache level `L`. The eviction policy is to give
+lower priority to data cached using this property versus the `cached`
+property.
+a|
+[source]
+----
+cache_control_write<cache_control_write_type::uncached, L>
+----
+|
+This property requests that writes to memory through the `annotated_ptr`
+should not cache the data at level `L` in the memory hierarchy.
+a|
+[source]
+----
+cache_control_write<cache_control_write_type::write_through, L>
+----
+|
+This property requests that writes to memory through the `annotated_ptr`
+should immediately write the data to the next-level cache after `L`
+and mark the cache line at level `L` as "not dirty".
+a|
+[source]
+----
+cache_control_write<cache_control_write_type::write_back, L>
+----
+|
+This property requests that writes to memory through the `annotated_ptr`
+should write the data into the cache at level `L` and mark the cache line as
+"dirty". Upon eviction, "dirty" data will be written into the cache at
+level higher than `L`.
+a|
+[source]
+----
+cache_control_write<cache_control_write_type::streaming, L>
+----
+|
+This property is the same as `write_through`, but requests use of a
+policy that gives lower priority to data in the cache present
+via a `streaming` cache control.
+|====
+--
+
+==== Assertions by the application
+These properties are assertions by the application, promising that the
+application accesses memory in a certain way. Care must be taken when
+using these properties because they can lead to undefined behavior if
+they are misused.
+
+--
+[options="header", cols="3,1"]
+|====
+| Property | Description
+a|
+[source]
+----
+cache_control_read<cache_control_read_type::invalidate_after_read, L>
+----
+|
+This property asserts that the cache line into which data is loaded
+from memory through the `annotated_ptr` will not be read again
+until it is overwritten. Therefore the load operation can invalidate
+the cache line and discard "dirty" data. If the assertion is violated 
+(i.e., the cache line is read again) then the behavior is undefined.
+a|
+[source]
+----
+cache_control_read<cache_control_read_type::const_cached, L>
+----
+|
+This property asserts that the cache line containing the data
+loaded from memory through the `annotated_ptr` will not be written
+until kernel execution is completed.
+If the assertion is violated (the cache line is written), the behavior
+is undefined.
+|====
+--
+
+== Implementation notes
+
+It is intended that the SYCL cache control properties will be used by the
+compiler to generate SPIR-V cache control operations. Alternatively, the
+properties could be implemented by generating intrinsic function calls
+that match the cache control types.
```

Comment:
Just out of curiosity, are we thinking that we would use intrinsics only for the targets that don't use SPIR-V?  For example, we obviously can't use the SPIR-V cache control extension for the CUDA backend.

Reply:
I think we would use SPIR-V when we can and if the SPIR-V extension is not yet implemented or is not going to be available, and intrinsics are available, we could use those.